chain/neutrino.go

```diff
@@ -45,20 +52,45 @@ type NeutrinoClient struct {
 	finished   bool
 	isRescan   bool
 
+	// The clientMtx synchronizes access to the state variables of the client.
+	// TODO(mstreet3): Currently the clientMtx synchronizes access to the
+	// rescanQuit and rescanErr channels, which cancel the current rescan
+	// goroutine when closed and is updated each time a new rescan goroutine
+	// is created, respectively.  All state related to the rescan goroutine
+	// should ideally be synchronized by the same lock or via some other
+	// shared mechanism.
 	clientMtx sync.Mutex
 }
 
 // NewNeutrinoClient creates a new NeutrinoClient struct with a backing
 // ChainService.
 func NewNeutrinoClient(chainParams *chaincfg.Params,
-	chainService *neutrino.ChainService) *NeutrinoClient {
+	chainService neutrinoChainService,
+	newRescan newRescanFunc) *NeutrinoClient {
```

Comment:
I see your point about changing the NewNeutrinoClient constructor. It's feasible to let this constructor define the rescan constructor internally. 

I don't see an issue with exporting the neutrinoChainService interface. I left it private assuming all consumers would just use `*neutrino.ChainService` (I.e., no other concrete implementations exist or plan to exist).

Reply:
> I left it private assuming all consumers would just use `*neutrino.ChainService` (I.e., no other concrete implementations exist or plan to exist).

That's what I gathered.  In that case, I don't see a reason to use the interface in the constructor's input args.  Just changing the `CS` field type would suffice since the tests aren't using the constructor anyway. Even that is a breaking API change, but not as disruptive I think since that field probably shouldn't have been exported anyway.

Reply:
Right right, let me give it a try. Would be good to reduce the surface area of the change.